scripts/sign-macos-pkg.sh

```diff
@@ -3,18 +3,50 @@
 set -e
 
 [[ $(uname) != 'Darwin' ]] && { echo 'This only works on macOS.' >&2; exit 1; }
-[[ $# -lt 2 ]] && { echo 'sign-macos-bundle.sh <file_to_sign> <sign_identity>' >&2; exit 1; }
+[[ $# -lt 3 ]] && { echo 'sign-macos-bundle.sh <method> <file_to_sign> <sign_identity>' >&2; exit 1; }
 
-# First is the target file/directory to sign
-TARGET="${1}"
-# Second argument is the signing identity
-CODESIGN_ID="${2}"
+# First is the sign method (frameworks, app or dmg)
+METHOD="${1}"
+# Second is the target file/directory to sign
+TARGET="${2}"
+# Third argument is the signing identity
+CODESIGN_ID="${3}"
 # Rest are extra command line flags for codesign
-shift 2
-CODESIGN_OPTS_EXTRA=("${@}")
+CODESIGN_OPTS_EXTRA=("${@:4}")
 
 [[ ! -e "${TARGET}" ]] && { echo 'Target file does not exist.' >&2; exit 1; }
 
+function sign_frameworks {
+    echo -e "\n### Sign frameworks"
+    find "$TARGET" -type f \( -name "*.dylib" -o -name "*.so" -o -name "*.framework" \) -print0 | while IFS= read -r -d '' file; do
+        CODESIGN_OPTS=(
+            "--sign ${CODESIGN_ID}"
+            "--verbose=4"
+            "--force"
+        )
```

Comment:
What is happening her? Why are we needlessly repeating the same sections like `CODESIGN_OPTS` or `CODESIGN_OPTS`? This could be done in a generic way.